knimin/lib/data_access.py

```diff
@@ -3191,6 +3186,252 @@ def delete_sample_plate(self, id):
             TRN.add(sql, [id])
             TRN.execute()
 
+    def get_id_by_name(self, field, name):
```

Comment:
There are quite a few Python codes added that are dedicated to the interface. They are useless without the interface. None of them is tested. As I mentioned previously in email, they are "crappy" codes. In the past three days I spent time to make them less crappy. Before changing or testing the codes there is a strategic question to be solved (please see bullet point #4 in my master comment to this PR).